chainntnfs/bitcoindnotify/bitcoind_test.go

```diff
@@ -0,0 +1,350 @@
+		tx, err := h.Node.GetRawTransaction(txid)
+		if err != nil {
+			jsonErr, ok := err.(*btcjson.RPCError)
+			if ok && jsonErr.Code == btcjson.ErrRPCNoTxInfo {
+				continue
+			}
+			return err
+		}
+
+		if tx != nil && tx.Hash().IsEqual(txid) {
+			break
+		}
+	}
+
+	return nil
+}
+
+func createSpendableOutput(t *testing.T, miner *rpctest.Harness) (*wire.OutPoint, []byte) {
+	// Create a transaction that only has one output, the one destined for
+	// the recipient.
+	script, err := txscript.PayToAddrScript(testAddr)
+	if err != nil {
+		t.Fatalf("unable to create p2pkh script: %v", err)
+	}
+	output := &wire.TxOut{Value: 2e8, PkScript: script}
+	txid, err := miner.SendOutputsWithoutChange([]*wire.TxOut{output}, 10)
+	if err != nil {
+		t.Fatalf("unable to create tx: %v", err)
+	}
+
+	// Mine the transaction to mark the output as spendable.
+	if err := waitForMempoolTx(miner, txid); err != nil {
+		t.Fatalf("tx not relayed to miner: %v", err)
+	}
+	if _, err := miner.Node.Generate(1); err != nil {
+		t.Fatalf("unable to generate single block: %v", err)
+	}
+
+	return wire.NewOutPoint(txid, 0), script
+}
+
+func createSpendTx(t *testing.T, outpoint *wire.OutPoint, pkScript []byte) *wire.MsgTx {
+	spendingTx := wire.NewMsgTx(1)
+	spendingTx.AddTxIn(&wire.TxIn{PreviousOutPoint: *outpoint})
+	spendingTx.AddTxOut(&wire.TxOut{Value: 1e8, PkScript: pkScript})
+
+	sigScript, err := txscript.SignatureScript(
+		spendingTx, 0, pkScript, txscript.SigHashAll, privKey, true,
+	)
+	if err != nil {
+		t.Fatalf("unable to sign tx: %v", err)
+	}
+	spendingTx.TxIn[0].SignatureScript = sigScript
+
+	return spendingTx
+}
+
+// TestHistoricalConfDetailsTxIndex ensures that we correctly retrieve
+// historical confirmation details using the backend node's txindex.
+func TestHistoricalConfDetailsTxIndex(t *testing.T) {
+	miner := setUpMiner(t)
+	defer miner.TearDown()
+
+	rpcPort, zmqPath, cleanUp := setUpBackend(t, miner.P2PAddress(), true)
+	defer cleanUp()
+
+	notifier := setUpNotifier(t, rpcPort, zmqPath)
+	defer notifier.Stop()
+
+	syncNotifierWithMiner(t, notifier, miner)
+
+	// A transaction unknown to the node should not be found within the
+	// txindex even if it is enabled, so we should not proceed with any
+	// fallback methods.
+	_, txStatus, err := notifier.historicalConfDetails(&zeroHash, 0, 0)
+	if err != nil {
+		t.Fatalf("unable to retrieve historical conf details: %v", err)
+	}
+
+	switch txStatus {
+	case txNotFoundIndex:
+	case txNotFoundManually:
+		t.Fatal("should not have proceeded with fallback method, but did")
+	default:
+		t.Fatal("should not have found non-existent transaction, but did")
+	}
+
+	// Now, we'll create a test transaction, confirm it, and attempt to
+	// retrieve its confirmation details.
+	txid, err := getTestTxId(miner)
+	if err != nil {
+		t.Fatalf("unable to create tx: %v", err)
+	}
+	if err := waitForMempoolTx(miner, txid); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := miner.Node.Generate(1); err != nil {
```

Comment:
Check the mempool case before generating a block?

Reply:
Can't do so for `bitcoind` at the moment until https://github.com/btcsuite/btcd/pull/1222 is in as we'll need to wait for `bitcoind` to detect the transaction in its mempool.